### ISSUE TITLE
Receiving event for passing each step of process

### ISSUE THREAD

**Issue by aramashkan** [apache/incubator-kie-kogito-runtimes#2155]
### Describe the bug

Event addon producing events that contains collected info about passed steps. Main problem that part about variables contains only final state. For example process like this:

![Screenshot 2022-04-19 at 13 31 05](https://user-images.githubusercontent.com/49194616/163985268-57cb229f-ddb0-4898-8c9e-e9421be6bc99.png)

will generate event like this:

```
{
  "id": "19adfd62-2bc3-49bc-84b5-7e90b409692d",
  "source": "/travellers",
  "type": "ProcessInstanceEvent",
  "time": "2022-04-18T17:44:31.442171+03:00",
  "data": {
    "id": "e47d28c3-c92c-43e4-894c-a2e8a046405d",
    "parentInstanceId": null,
    "rootInstanceId": null,
    "processId": "travellers",
    "rootProcessId": null,
    "processName": "travellers",
    "startDate": "2022-04-18T17:44:31.432+03:00",
    "endDate": "2022-04-18T17:44:31.440+03:00",
    "state": 2,
    "businessKey": null,
    "nodeInstances": [
      {
        "id": "a7baf232-637b-4d43-b219-6d4343997812",
        "nodeId": "4",
        "nodeDefinitionId": "_0528A25B-7F32-4721-9B3C-86E6329EA06E",
        "nodeName": "Done",
        "nodeType": "EndNode",
        "triggerTime": "2022-04-18T17:44:31.439+03:00",
        "leaveTime": "2022-04-18T17:44:31.439+03:00"
      },
      {
        "id": "132fdc40-9aab-4d07-8996-b8f633112759",
        "nodeId": "1",
        "nodeDefinitionId": "_E03285FC-0EB4-452A-8746-0BAF0A3A93E3",
        "nodeName": "Service Task 3",
        "nodeType": "WorkItemNode",
        "triggerTime": "2022-04-18T17:44:31.437+03:00",
        "leaveTime": "2022-04-18T17:44:31.439+03:00"
      },
      {
        "id": "b942e6cc-1e33-4849-9d16-4ad914e5df30",
        "nodeId": "2",
        "nodeDefinitionId": "_8A6DAA0B-DDBF-4A7D-BBF1-CDC2869E05DD",
        "nodeName": "Service Task 2",
        "nodeType": "WorkItemNode",
        "triggerTime": "2022-04-18T17:44:31.437+03:00",
        "leaveTime": "2022-04-18T17:44:31.437+03:00"
      },
      {
        "id": "72369770-82f5-4d94-9a8c-e40a9710e7b5",
        "nodeId": "3",
        "nodeDefinitionId": "_0E0EAFCE-F990-4509-9AC4-445563CF3690",
        "nodeName": "Service Task 1",
        "nodeType": "WorkItemNode",
        "triggerTime": "2022-04-18T17:44:31.432+03:00",
        "leaveTime": "2022-04-18T17:44:31.437+03:00"
      },
      {
        "id": "378782da-33ef-48dc-be74-e94f5f4874d5",
        "nodeId": "5",
        "nodeDefinitionId": "_4D347C24-28B2-43D6-98B6-A0BEFACD0586",
        "nodeName": "Start Event",
        "nodeType": "StartNode",
        "triggerTime": "2022-04-18T17:44:31.432+03:00",
        "leaveTime": "2022-04-18T17:44:31.432+03:00"
      }
    ],
    "variables": {
      "result": "service call 3"
    },
    "error": null,
    "roles": null,
    "milestones": []
  },
  "specversion": "1.0",
  "kogitoprocinstanceid": "e47d28c3-c92c-43e4-894c-a2e8a046405d",
  "kogitoprocid": "travellers",
  "kogitoaddons": "process-management",
  "kogitousertaskist": "2"
}
```

### Expected behavior

Need possibility to receive event without loosing info about variables state after each step.
Is there any way to achieve this ?

### Output of `java -version`

openjdk version "17.0.1" 2021-10-19 LTS OpenJDK Runtime Environment Zulu17.30+15-CA (build 17.0.1+12-LTS) OpenJDK 64-Bit Server VM Zulu17.30+15-CA (build 17.0.1+12-LTS, mixed mode, sharing)

### Kogito version or git rev (or at least Quarkus version if you are using Kogito via Quarkus platform BOM)

Tested on 1.8.0.Final and 1.20.0.Final 

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)

**Comment by fjtirado** [comment]
Per design, variables of ProcessInstanceEventType contains the current (what you are calling final, because yours is what we call a straight through process, ie, one that does not contain any wait state ) value when the event is published. 
But there should be there events of type VariableInstanceEvent reflecting variable changes. Can you see them?

**Comment by fjtirado** [comment]
I checked that not event is being published to kogito-variables-events topic. 
Thats wrong and should be fixed with https://github.com/kiegroup/kogito-runtimes/pull/2158
Once this is merged you should, in that topic, these events
```

{"id":"ca514c06-e987-4815-84c8-50db8bc32405","source":"/orders","type":"VariableInstanceEvent","time":"2022-04-20T18:13:46.569198+02:00","data":{"variableName":"order","variableValue":{"orderNumber":"12345","shipped":false,"total":0.8743414772726855},"variablePreviousValue":null,"changeDate":"2022-04-20T18:13:46.548+02:00","changedByNodeId":null,"changedByNodeName":null,"changedByNodeType":null,"changedByUser":null,"processInstanceId":"07d51191-0921-4547-a0fe-38f515915c2a","rootProcessInstanceId":null,"processId":"demo.orders","rootProcessId":null},"kogitoVariableName":"order","specversion":"1.0","kogitoprocinstanceid":"07d51191-0921-4547-a0fe-38f515915c2a","kogitoprocid":"demo.orders","kogitoaddons":"process-management"}
{"id":"fb3544db-1326-490c-9622-9ce686736919","source":"/orderItems","type":"VariableInstanceEvent","time":"2022-04-20T18:13:46.569219+02:00","data":{"variableName":"order","variableValue":{"orderNumber":"12345","shipped":false,"total":0.8743414772726855},"variablePreviousValue":null,"changeDate":"2022-04-20T18:13:46.554+02:00","changedByNodeId":null,"changedByNodeName":null,"changedByNodeType":null,"changedByUser":null,"processInstanceId":"56d49008-0ecf-4063-831e-54db8dbf119d","rootProcessInstanceId":"07d51191-0921-4547-a0fe-38f515915c2a","processId":"demo.orderItems","rootProcessId":"demo.orders"},"kogitoVariableName":"order","specversion":"1.0","kogitoprocinstanceid":"56d49008-0ecf-4063-831e-54db8dbf119d","kogitorootprociid":"07d51191-0921-4547-a0fe-38f515915c2a","kogitoprocid":"demo.orderItems","kogitorootprocid":"demo.orders","kogitoaddons":"process-management"}

```
